### PR TITLE
Compare VAE vs noise simulation

### PR DESCRIPTION
**Pull request by ajlee21**
**Question:** Can we separate between generic and specific genes by adding gaussian noise to simulate experiments? Does VAE approach recapitulate generic genes better than gaussian noise approach?

To answer this question we compared how well SOPHIE (VAE approach) could recapitulate manually curated generic genes (Crow et al.) compared to generic genes generated using noise approach

Results:
SOPHIE vs Crow et al
image

Noise vs Crow et al.
image

Looks like noise and VAE can both recapitulate generic genes, which is expected.

Template experiment vs Crow et al
image

* looks like template experiment already expresses generic genes, so adding a small amount of noise (Normal(0,2)) will still find these generic results. This is expected, given that generic genes are "generic" because they are found across many experiments

**Main takeaway:**
What we really want is to determine if SOPHIE can better **separate** between generic and specific genes. To do this, we would need a gold standard for what are specific genes for some experiment, which we do not have. So for now we will leave the experiment as is.

**Changes int his PR:**
The main change can be found in `explore_simulation_approach\`

Most of the code should look the same as what was previously reviewed. The main difference is:
* The simulation of the data in `1_simulate_and_identify_generic_genes_using_noise_approach.ipynb`
* `comparisons_against_template.ipynb`

### CODE REVIEW

File: explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+# Hypergeometric test:
+# Given N number of genes with K common genes in Crow et al.
+# SOPHIE identifies n genes as being common
+# What is the probability that k of the genes identified by SOPHIE
+# are also common in Crow et al.? What is the probability of drawing
+# k or more concordant genes?
+
+num_Crow_genes = shared_ranking.shape[0]
+num_generic_Crow_genes = shared_ranking.query(f"{ref_rank_col}>=80.0").shape[0]
+num_generic_SOPHIE_genes = shared_ranking[
```
Comment (jjc2718) [inline explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py:323]: Are "SOPHIE genes" here genes identified as generic after adding Gaussian noise? I would think SOPHIE would involve your VAE/latent space shift method, but maybe I'm wrong.

**Review by jjc2718 — APPROVED** [review]
Looks good - just a few comments.

Conceptual question: I'm still just a bit confused about why you'd expect adding Gaussian noise to identify generic genes without needing to train a VAE. Is it because you have many experiments in your dataset/compendium already for this experiment, so you don't need a VAE to generate more experiments?

If this is the case, what does the VAE approach add? Maybe it allows you to get the generic genes using less starting data (because you can generate more)? Or is the sigmoid curve showing you that the VAE can detect more generic genes?

**Comment by ajlee21** [comment]
> Conceptual question: I'm still just a bit confused about why you'd expect adding Gaussian noise to identify generic genes without needing to train a VAE. Is it because you have many experiments in your dataset/compendium already in this case, so you don't need a VAE to generate more experiments?
> 
> If this is the case, what does the VAE approach add? Maybe it allows you to get the generic genes using less starting data (because you can generate more)? Or is the sigmoid curve showing you that the VAE can detect more generic genes?

Good questions. I will try to clarify my comments in the notebook. So the reasons that we think that generic genes are found by both the VAE approach and this noise approach is because they are "generic". So these generic signals are already found to exist across many experiments and by adding noise to the experiments we are disrupting that signal a bit but its still there.

The benefit to using a VAE, presumably, is that the VAE will allow us to identify those specific genes by generating different types of experiments, where as the noise approach is limited to generating the same experiment but with different amounts of noise added.

File: explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+processed_template_filename = params["processed_template_filename"]
+num_runs = params["num_simulated"]
+col_to_rank_genes = params["rank_genes_by"]
+count_threshold = params["count_threshold"]
+logFC_name = params["DE_logFC_name"]
+pvalue_name = params["DE_pvalue_name"]
+
+# Set mean and variance for noise distribution
+mu = 0
+sigma = 1000
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py:73]: Yes, this was a parameter I varied. 

I'll update this text so it's correct. Thank you for checking

File: explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+plt.title("Std gene expression for template experiment")
+
+# ## Quick check
+#
+# Check that we are producing distinct simulated experiments (i.e. that we are not getting the same values for each simulated experiment)
+
+mapped_template.head()
+
+# +
+simulated_data_filename_0 = os.path.join(
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py:141]: These numbers are identifiers for the different simulated experiments generated. I randomly chose 2 different simulated experiments to look at. I'll add a note here for this.

File: explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py (hunk 1 above)
Comment (ajlee21) [reply explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py:323]: This is definitely a "copy and paste" error:) I will fix this

**Comment by jjc2718** [comment]
> Good questions. I will try to clarify my comments in the notebook. So the reasons that we think that generic genes are found by both the VAE approach and this noise approach is because they are "generic". So these generic signals are already found to exist across many experiments and by adding noise to the experiments we are disrupting that signal a bit but its still there.

Got it. So if you just ran DE on the dataset without adding any noise, you'd find the same (or similar) genes? Maybe you already tried this and I'm forgetting it. 

> The benefit to using a VAE, presumably, is that the VAE will allow us to identify those specific genes by generating different types of experiments, where as the noise approach is limited to generating the same experiment but with different amounts of noise added.

Makes sense. Is that something you're planning to look at in the future?

**Comment by ajlee21** [comment]
> Got it. So if you just ran DE on the dataset without adding any noise, you'd find the same (or similar) genes? Maybe you already tried this and I'm forgetting it.

Yes this is the last figure in the description above "Template experiment vs Crow et al"

> Makes sense. Is that something you're planning to look at in the future?

This is something where we don't have the data to perform. We would need a dataset that has a gold standard set of specific genes. Perhaps there is a work around experiment we could perform, but for now we are leaving this experiment for after the manuscript is submitted/if time permitted. Definitely a very interesting experiment to perform

**Pull request merged**